internal/profiles/service.go

```diff
@@ -564,7 +564,7 @@ func upsertRuleInstances(
 	updatedIDs := make([]uuid.UUID, len(newRules))
 	for i, rule := range newRules {
 		// TODO: Clean up this logic once we no longer have to support the old tables.
-		entityRuleTuple, ok := rulesInProf[RuleTypeAndNamePair{
+		ruleIDAndName, ok := rulesInProf[RuleTypeAndNamePair{
 			RuleType: rule.Type,
 			RuleName: rule.Name,
```

Comment:
I'll believe you, but what I'm seeing is that:

1. `CreateProfile` calls `PopulateRuleNames` on `profile` on `service.go:129`, passing a _pointer_ (in/out value).
2. `PopulateRuleNames` iterates over Repository, BuildEnvironment, PullRequest, Artifact slices (which are `[]*pb.Profile_Rule`), calling a function which sets the `r.Name` parameter on the pointed-to object.
3. `createProfileRulesForEntity` builds a map from the Repository, Artifact, etc slices of `[]*pb.Profile_Rule` (all 8 sub-fields, different from PopulateRuleNames), and then calls `createProfileRulesForEntity` on each of the lists (`service.go:168`).  These lists will have had `r.Name` populated in (2).
4. `createProfileRulesForEntity` will call `upsertRuleInstances` at `service.go:375`, passing in the slice from (3) AND marshalling the `[]*pb.Profile_Rule` into JSON (using `encoding/json` and not `protojson`!) and storing them in `CreateProfileForEntity` at `service.go:393`.
5. `upsertRuleInstances` iterates over the `[]*pb.Profile_Rule` slice from (3),  looking up `(r.Type, r.Name)` in `entityRuleTuple` and then calling `qtx.UpsertRuleInstance` in the database with `r.Name` on `service.go:590`.

I'm being a bit stubborn because I'm concern about the database stores in (4) and (6) -- I'd prefer not to do another cleanup like `000017`.

Reply:
Thanks, I see what you mean now.
I initially thought we should not return the defaulted rule name, when getting / listing profiles because the output should conform to the user input, i.e if the user didn't set a rule `name` override in the profile then they shouldn't see a `name` when fetching the profile.
But, I can see that's not how it used to work previously. Instead we did always return the defaulted name, even if it did not come from the profile itself. I'll update the code to continue to behave this way.